models/channel_helpers.go

```diff
@@ -0,0 +1,295 @@
+	}
+
+	channel := &Channel{ID: id}
+	switch err := q.Reload(channel); err {
+	case nil:
+		return channel, nil
+	case reform.ErrNoRows:
+		return nil, status.Errorf(codes.NotFound, "Channel with ID %q not found.", id)
+	default:
+		return nil, errors.WithStack(err)
+	}
+}
+
+// CreateChannelParams are params for creating new channel.
+type CreateChannelParams struct {
+	Summary string
+
+	EmailConfig     *EmailConfig
+	PagerDutyConfig *PagerDutyConfig
+	SlackConfig     *SlackConfig
+	WebHookConfig   *WebHookConfig
+
+	Disabled bool
+}
+
+// CreateChannel persists notification channel.
+func CreateChannel(q *reform.Querier, params *CreateChannelParams) (*Channel, error) {
+	id := "/channel_id/" + uuid.New().String()
+
+	if err := checkUniqueChannelID(q, id); err != nil {
+		return nil, err
+	}
+
+	if params.Summary == "" {
+		return nil, status.Error(codes.InvalidArgument, "Channel summary can't be empty.")
+	}
+
+	row := &Channel{
+		ID:       id,
+		Summary:  params.Summary,
+		Disabled: params.Disabled,
+	}
+
+	if params.EmailConfig != nil {
+		if err := checkEmailConfig(params.EmailConfig); err != nil {
+			return nil, err
+		}
+		row.Type = Email
+		row.EmailConfig = params.EmailConfig
+	}
+
+	if params.PagerDutyConfig != nil {
+		if row.Type != "" {
+			return nil, invalidConfigurationError
+		}
+
+		if err := checkPagerDutyConfig(params.PagerDutyConfig); err != nil {
+			return nil, err
+		}
+		row.Type = PagerDuty
+		row.PagerDutyConfig = params.PagerDutyConfig
+	}
+
+	if params.SlackConfig != nil {
+		if row.Type != "" {
+			return nil, invalidConfigurationError
+		}
+		if err := checkSlackConfig(params.SlackConfig); err != nil {
+			return nil, err
+		}
+		row.Type = Slack
+		row.SlackConfig = params.SlackConfig
+	}
+
+	if params.WebHookConfig != nil {
+		if row.Type != "" {
+			return nil, invalidConfigurationError
+		}
+		if err := checkWebHookConfig(params.WebHookConfig); err != nil {
+			return nil, err
+		}
+		row.Type = WebHook
+		row.WebHookConfig = params.WebHookConfig
+	}
+
+	if row.Type == "" {
+		return nil, status.Error(codes.InvalidArgument, "Missing channel configuration.")
+	}
+
+	err := q.Insert(row)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create notifications channel")
+	}
+
+	return row, nil
+}
+
+// ChangeChannelParams is params for changing existing channel.
+type ChangeChannelParams struct {
+	EmailConfig     *EmailConfig
+	PagerDutyConfig *PagerDutyConfig
+	SlackConfig     *SlackConfig
+	WebHookConfig   *WebHookConfig
+
+	Disabled bool
+}
+
+// ChangeChannel updates existing notifications channel.
+func ChangeChannel(q *reform.Querier, channelID string, params *ChangeChannelParams) (*Channel, error) {
+	row, err := FindChannelByID(q, channelID)
+	if err != nil {
+		return nil, err
+	}
+
+	// remove previous configuration
+	row.EmailConfig = nil
+	row.PagerDutyConfig = nil
+	row.SlackConfig = nil
+	row.WebHookConfig = nil
+
+	if params.EmailConfig != nil {
```

Comment:
this part of the code which sets one of the four configs is same as the logic used in `CreateChannel`, can't we extract it to a function `setConfig(row, params)`?

Reply:
`CreateChannel` and `ChangeChannel` use different param models.